mdspyutils/mdspyutils.py

```diff
@@ -32,13 +32,78 @@ def count_points(file_name: str = None) -> pd.DataFrame:
     return None
 
 
-def check_mechanics(file_name: str) -> bool:
-    """[summary]
+def check_repo_link(file_name: str) -> bool:
+    """ Check whether the user has included the github repo link in his 
+        repository
 
     Args:
-        file_name (str): [description]
+        file_name (str): A path or list of paths to MDS lab files (either
+            .ipynb or .Rmd). If left blank, the function will recursively
+            search for all labs in the working directory based on the file
+            extension.
+
+    Returns:
+        bool: a boolean output 
+
+    Example:
+        check_repo_link()    
+    """
+    return None
+
+
+def check_lat_version(file_name: str) -> bool:
+    """ Check whether the user has pushed the latest version in his 
+        repository
+
+    Args:
+        file_name (str): A path or list of paths to MDS lab files (either
+            .ipynb or .Rmd). If left blank, the function will recursively
+            search for all labs in the working directory based on the file
+            extension.
+
+    Returns:
+        bool: a boolean output
+
+    Example:
+        check_lat_version() 
+    """
+    return None
+
+
+def check_commits(file_name: str) -> bool:
+    """ Check whether the user has atleast three commits
+
+    Args:
+        file_name (str): A path or list of paths to MDS lab files (either
+            .ipynb or .Rmd). If left blank, the function will recursively
+            search for all labs in the working directory based on the file
+            extension.
+
+    Returns:
+        bool: a boolean output
+
+    Example:
+        check_commits()      
+    """
+    return None
+
+
+def check_mechanics(file_name: str) -> NoneType:
+    """ It is a wrapper function which calls the other functions - check_repo_link, 
+        check_commits, check_lat_version 
```

Comment:
Since this is a user-facing function, I think it might be better to list out what checks are being done. Something like:

> """Performs Mechanics Checks on a MDS Lab
> 
> This function check that you have a Github repo link, that you have pushed your latest commit, and that you have at least three commit messages authored by you in your history.